open-sphere-plugins/csv/src/main/java/io/opensphere/csv/parse/CSVDataElementProvider.java

```diff
@@ -49,6 +49,7 @@ public CSVDataElementProvider(Toolbox tb, DefaultDataTypeInfo dti, Runnable conf
             boolean useDeterminedDataTypes, boolean useDynamicEnumerations)
         throws FileNotFoundException, IOException
     {
+        super(dti.getTypeKey());

```

Comment:
`myFileSource` needs to be set before this call due to `getParseParams()`, maybe move that to a protected field and set it in the superclass constructor as well? Or define getParseParams there instead.